pkgs/applications/networking/instant-messengers/discord/openasar.nix

```diff
@@ -0,0 +1,63 @@
+{ lib, stdenv, fetchFromGitHub, fetchYarnDeps, nodejs-16_x, yarn, bash, fixup_yarn_lock }:
+
+stdenv.mkDerivation rec {
+  version = "unstable-2022-06-10";
+  pname = "openasar";
+
+  src = fetchFromGitHub {
+    owner = "GooseMod";
+    repo = "OpenAsar";
+    rev = "c6f2f5eb7827fea14cb4c54345af8ff6858c633a";
+    sha256 = "m6e/WKGgkR8vjKcHSNdWE25MmDQM1Z3kgB24OJgbw/w=";
+  };
+
+  nativeBuildInputs = [ nodejs-16_x yarn bash fixup_yarn_lock ];
+
+  yarnOfflineCache = fetchYarnDeps {
+    yarnLock = ./openasar/yarn.lock;
+    sha256 = "I892xicWoGyogpXb76FNlv/hrxOU4bk5DK54+V3Abas=";
+  };
+
+  configurePhase = ''
+    runHook preConfigure
+
+    export HOME=$(mktemp -d)
+    cp ${./openasar/package.json} ./package.json
+    cp -L --no-preserve mode ${./openasar/yarn.lock} ./yarn.lock
+    yarn config --offline set yarn-offline-mirror ${yarnOfflineCache}
+    ${fixup_yarn_lock}/bin/fixup_yarn_lock yarn.lock
+    yarn install --offline --frozen-lockfile --ignore-platform --ignore-scripts --no-progress --non-interactive
+    patchShebangs node_modules/
+
+    runHook postConfigure
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    bash scripts/injectPolyfills.sh
+    sed -i -e "s/nightly/${version}/" src/index.js
+    node scripts/strip.js
+    yarn exec --offline -- asar pack src app.asar
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install app.asar $out
+
+    runHook postInstall
+  '';
+
+  doCheck = false;
+
+  meta = with lib; {
```

Comment:
I've added `meta` even if this derivation won't hit top-level all packages (it only lives within discord's install phase) since it has its own project/license/homepage.